### DISCUSSION TITLE
How to find zero-bonds from Smiles?

### DISCUSSION THREAD

**Discussion by amitlevy** [rdkit/rdkit#4400] [Q&A]
We are currently in the process of replacing the python module PySmiles in our project with rdkit. What we want to do is convert SMILES strings into a single connected graph that represents the molecule. But unlike PySmiles, where all the resulting molecules are connected graphs, from rdkit we seem to get disjoint graphs (after turning every atom into a node and every bond into an edge). We believe the issue is that when running GetBonds() on the result of MolFromSmiles(), zero-bonds don't appear. Is there a way we can fix this issue so we can add the zero-bonds to the graph?

**Comment by greglandrum** [comment]
There's no such thing as a "zero bond" in SMILES. I guess you are referring to something added by PySMILES when there's a "." in the SMILES string. The ".", however,  represents a disconnection point. You can refer to the "Disconnected structures" section of the Daylight SMILES reference for more info on this: https://www.daylight.com/dayhtml/doc/theory/theory.smiles.html

If you for some reason have SMILES where you believe that the positions of those diconnections are important (and this is definitely not generally true), then you could take advantage of an RDKit extension to SMILES and replace all of the `.` characters in the SMILES with `~` characters. This will add zero-order bonds to the molecule.
